HomeWork/src/main/java/animals/Cat.java

```diff
@@ -0,0 +1,28 @@
+package animals;
+
+public class Cat extends Animal {
+
+    private static int catsCounter;
+
+    public Cat(String name, int runRestriction, int swimRestriction) {
+        super(name, runRestriction, swimRestriction);
+        catsCounter++;
+    }
+
+    public static int getCatsCounter() {
+        return catsCounter;
+    }
+
+    @Override
+    public void run(int lengthBarrier) {
+        super.run(lengthBarrier);
```

Comment:
Why do you call `super.run()` before any check? remove this call as you shouldn't print a message `Animal running` before your logic below!